state-chain/pallets/cf-swapping/src/tests/fill_or_kill.rs

```diff
@@ -3,11 +3,11 @@ use super::*;
 const BROKER_FEE: AssetAmount = INPUT_AMOUNT * BROKER_FEE_BPS as u128 / 10_000;
 
 fn fok_swap(refund_params: Option<TestRefundParams>) -> TestSwapParams {
-	params(None, refund_params, false)
+	TestSwapParams::new(None, refund_params, false)
```

Comment:
Why not create a constructor/builder instead of using a free function?

```
// Fok without ccm
TestSwapParams::fok(refund_params);
// or
TestSwapParams::new().with_fok(refund_params);

// Fok with ccm
TestSwapParams::fok_with_ccm(refund_params);
// or
TestSwapParams::new().with_fok(refund_params).with_ccm();
```


This can be extended to dca: we need fok params for dca, and we also want to test ccm with dca.

Reply:
didn't seem worth it. It is already quite easy and clean to just provide None if not using it.

Reply:
I don't mind the builder pattern. I somewhat agree with Jamie that it might not be worth it at this stage, but we will likely have more paramters in the future, and I think it is a change we will eventually want to make at some point.